tsconfig.unit.json

```diff
@@ -1,7 +1,8 @@
 {
   "extends": "./tsconfig.json",
   "compilerOptions": {
-    "lib": ["es2019", "dom"],
+    "lib": ["es2020", "dom"],
+    "moduleResolution": "bundler",
```

Comment:
There is an issue when running type checks for test files. Vitest does not pass type checks unless `moduleResolution: bundler` is used: https://github.com/vitest-dev/vitest/issues/4567

Needed to upgrade typescript for this as well